src/ripple/app/misc/ValidatorList.h

```diff
@@ -20,102 +20,424 @@
+        @param configKeys List of trusted keys from config. Each entry consists
+        of a base58 encoded validation public key, optionally followed by a
+        comment.
+
+        @param siteURIs List of URIs to fetch published validator lists
+
+        @param publisherKeys List of trusted publisher public keys. Each entry
+        contains a base58 encoded account public key.
+
+        @param configManifest Base64 encoded manifest for local node's signing
+        public key
+
+        @par Thread Safety
+
+        May be called concurrently
+
+        @return `false` if an entry is invalid or unparsable
     */
-    boost::optional<std::string>
-    member (
-        PublicKey const& identity) const;
+    bool
+    load (
+        PublicKey const& localSigningKey,
+        std::vector<std::string> const& configKeys,
+        std::vector<std::string> const& siteURIs,
+        std::vector<std::string> const& publisherKeys,
+        std::vector<std::string> const& configManifest);
+
+    /** Start fetching lists from sites
+
+        This does nothing if list fetching has already started
+
+        @par Thread Safety
+
+        May be called concurrently
+    */
+    void
+    start ();
+
+    /** Wait for current fetches from sites to complete
+
+        @par Thread Safety
+
+        May be called concurrently
+    */
+    void
+    join ();
+
+    /** Stop fetching lists from sites
+
+        This blocks until list fetching has stopped
+
+        @par Thread Safety
+
+        May be called concurrently
+    */
+    void
+    stop ();
+
+    /** Update trusted keys
+
+        Reset the trusted keys based on latest manifests, received validations,
+        and lists from sites.
+
+        @param seenValidators Set of public keys used to sign recently
+        received validations
+
+        @par Thread Safety
+
+        May be called concurrently
+    */
+    template<class KeySet>
+    void
+    onConsensusStart (
+        KeySet const& seenValidators);
+
+    /** Determine the quorum for the trusted list size
+
+        The quorum is the minimum number of validations needed for a ledger to
+        be fully validated. It is primarily dependent on the number of trusted
+        validation keys.
+
+        @param nTrusted Number of trusted validation public keys
+
+        @param nListed Number of keys included on any recommended lists
 
-    /** Determines whether a node is in the UNL */
+        @return quorum value
+    */
+    static std::size_t
+    calcQuorum (
+        std::uint32_t nTrusted,
+        std::uint32_t nListed);
+
+    /** Get quorum value for current trusted key set
+
+        The quorum is the minimum number of validations needed for a ledger to
+        be fully validated. It can change when the set of trusted validation
+        keys is updated (at the start of each consensus round) and primarily
+        depends on the number of trusted keys.
+
+        @par Thread Safety
+
+        May be called concurrently
```

Comment:
You're right. One option could be to add the mutex lock to `quorum()` which would allow it to return the latest quorum if one is already in the works in `onConsensusStart`.